lib/geocoder/lookups/google_places_search.rb

```diff
@@ -18,16 +18,42 @@ def supported_protocols
 
       private
 
+      def results(query)
+        super(query, 'candidates')
+      end
+
       def base_query_url(query)
-        "#{protocol}://maps.googleapis.com/maps/api/place/textsearch/json?"
+        "#{protocol}://maps.googleapis.com/maps/api/place/findplacefromtext/json?"
       end
 
       def query_url_google_params(query)
         {
-          query: query.text,
+          input: query.text,
+          inputtype: 'textquery',
+          fields: fields(query),
           language: query.language || configuration.language
         }
       end
+
+      def fields(query)
+        query_fields = query.options[:fields]
+        return format_fields(query_fields) if query_fields
+
+        default_fields
+      end
+
+      def default_fields
+        legacy = %w[id reference]
```

Comment:
`id` and `reference` fields were already available in the previous API response (Text search response), I added those 2 fields here to prevent any regression for developers already using the gem